casanovo/utils.py

```diff
@@ -66,3 +75,221 @@ def split_version(version: str) -> Tuple[str, str, str]:
     """
     version_regex = re.compile(r"(\d+)\.(\d+)\.*(\d*)(?:.dev\d+.+)?")
     return tuple(g for g in version_regex.match(version).groups())
+
+
+def get_score_bins(
+    results_table: pd.DataFrame, score_bins: List[float]
+) -> Dict[float, int]:
+    """
+    Get binned confidence scores
+
+    From a list of confidence scores, return a dictionary mapping each
+    confidence score to the number of spectra with a confidence greater
+    than or equal to it.
+
+    Parameters
+    ----------
+    results_table: pd.DataFrame
+        Parsed spectrum match table
+    score_bins: List[float]
+        Confidence scores to map
+
+    Returns
+    -------
+    score_bin_dict: Dict[float, int]
+        Dictionary mapping each confidence score to the number of spectra
+        with a confidence greater than or equal to it.
+    """
+    return {
+        score: (results_table["score"] >= score).sum() for score in score_bins
+    }
+
+
+def get_peptide_length_hist(peptide_lengths: np.ndarray) -> np.ndarray:
+    """
+    Get a matrix mapping each unique peptide length to its frequency
+
+    Parameters
+    ----------
+    peptide_lengths: np.ndarray
+        Numpy array containing the length of each sequence
+
+    Returns
+    -------
+    peptide_length_hist_matrix: np.ndarray
+        A matrix of size (n x 2) where each row vector is
+        (peptide_length, frequency)
+    """
+    frequncies = np.bincount(peptide_lengths).reshape((-1, 1))
```

Comment:
**typo:** `frequencies`

Reply:
I ended up just removing the histogram functionality entirely as discussed in the meeting.